dspace-api/src/main/java/org/dspace/content/integration/crosswalks/csl/DSpaceListItemDataProvider.java

```diff
@@ -146,6 +154,36 @@ public void processItem(Item item) {
         this.items.put(cslItemData.getId(), cslItemData);
     }
 
+    /**
+     * Overrides some properties of the current class depending on the configuration and type of the publication.
+     * 
+     * @param item The item to override fields for. Used to extract the publication type.
+     */
+    private void overrideFieldsForPubType(Item item) {
+        if ((item.getType() == Constants.ITEM) && (itemService.getEntityType(item).equals("Publication"))) {
+            // Get the publication type of the item.
+            String publicationType = getMetadataFirstValue(item, type);
+            if (typeSpecificFieldMap.containsKey(publicationType)) {
+                // Using the publication type get the fields to override.
+                typeSpecificFieldMap.get(publicationType).entrySet().forEach(field -> {
+                    try {
+                        // retrieve the corresponding field from the current class and try to set its value.
+                        Field classField = this.getClass().getDeclaredField(field.getKey());
+                        classField.setAccessible(true);
+                        classField.set(this, field.getValue());
+                    } catch (NoSuchFieldException e) {
+                        LOGGER.error(
+                            "Tried to override a non existing property of "
+                            + this.getClass() + ": [" + field.getKey() + "]"
+                        );
+                    } catch (IllegalAccessException ille) {
+                        LOGGER.error("Cannot access field [" + field.getKey() + "] on class " + this.getClass());
+                    }
+                });
```

Comment:
Why use `entrySet()` and not just iterate directly ? 
```java
typeSpecificFieldMap.get(publicationType).forEach((name, value) -> {
    try {
        Field classField = this.getClass().getDeclaredField(name);
        classField.setAccessible(true);
        classField.set(this, value);
   } ...
```
For logging you could use `blabla {}: {}` format :: 
```java
LOGGER.error("Tried to override a non existing property of {}: [{}]", this.getClass(), name);
```